### PR TITLE
Fix #1721

### LINKED ISSUE #1721: DynamicForm - more than 100 lookups and date format in lookup field
#### Category
[ ] Enhancement

[ x ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [ 3.16.0 ]

#### Expected / Desired Behavior / Question
Show all (or atleast the first 5000) lookup items in a lookup field. 
Display the date in a lookup field in a user friendly format instead an ISO-string.

#### Observed Behavior
Only 100 items are displayed in a lookup field. 
Date is shown as ISO-string in a lookup field.

#### Steps to Reproduce
Add a list with more than 100 items as a lookup.
Make the lookup display column a date column

### PR DESCRIPTION
**Pull request by nsoeth**
| Q               | A
| --------------- | ---
| Bug fix?        | [ ]
| New feature?    | [ ]
| New sample?      | [x]
| Related issues?  | fixes #X, partially #Y, mentioned in #Z

#### What's in this Pull Request?

This PR will fix #1721 

#### Guidance
- *You can delete this section when you are submitting the pull request.* 
- *Please update this PR information accordingly. We'll use this as part of our release notes in monthly communications.*
- *Please target your PR to **dev** branch.*

### CODE REVIEW

**Comment by michaelmaillot** [comment]
Hi @nsoeth,

Thanks for this PR!

Regarding the addition of _$top_ query operation, I'd suggest to make it available as optional because it depends on use case and maybe it would not fit for everyone to load 5000 items at once.

Since the DynamicForm lookup field relies on `ListItemPicker` control, and in order to be available as a field override prop for the DynamicForm, here are the different files to update according to me:

* [SPService.ts](https://github.com/pnp/sp-dev-fx-controls-react/blob/285cb32559ccb345e53deb388a3ad1aa9e97af6a/src/services/SPService.ts#L214)
* [IListItemPieckerProps](https://github.com/pnp/sp-dev-fx-controls-react/blob/285cb32559ccb345e53deb388a3ad1aa9e97af6a/src/controls/listItemPicker/IListItemPickerProps.ts#L5)
* [ListItemPicker](https://github.com/pnp/sp-dev-fx-controls-react/blob/285cb32559ccb345e53deb388a3ad1aa9e97af6a/src/controls/listItemPicker/ListItemPicker.tsx#L178) (`ListItemPicker`control doc will have to be updated as there will be a new prop available)
* [IDynamicFieldProps](https://github.com/pnp/sp-dev-fx-controls-react/blob/285cb32559ccb345e53deb388a3ad1aa9e97af6a/src/controls/dynamicForm/dynamicField/IDynamicFieldProps.ts#L8)
* [DynamicField](https://github.com/pnp/sp-dev-fx-controls-react/blob/285cb32559ccb345e53deb388a3ad1aa9e97af6a/src/controls/dynamicForm/dynamicField/DynamicField.tsx#L27)

Like this, the developer will be free to set the number of items he wants to get 🙂 

A suggestion of prop name could be `itemsQueryCountLimit` or `ODataTopCount` for example but feel free to suggest!

Let me know if something's not clear or if you need help.

**Comment by nsoeth** [comment]
Hi @michaelmaillot ,
Thanks for your suggestions, I updated the PR. 

Kind regards,
Niels

File: src/services/SPService.ts [hunk 1]
```diff
@@ -559,14 +565,24 @@ export default class SPService implements ISPService {
            const isArray = Array.isArray(result[fieldName]);
            //multiselect lookups are arrays
            if (isArray) {
-            result[fieldName].forEach(element => {
-              lookups.push({ key: element.ID, name: element[lookupFieldName || 'Title'] });
+            result[fieldName].forEach(element => {                
+              let value = element[fieldName][lookupFieldName || 'Title'];
```
Comment (michaelmaillot) [inline src/services/SPService.ts:569]: This should be `element[lookupFieldName || 'Title']` because you've already iterate into the `fieldName` data.

**Review by michaelmaillot — CHANGES_REQUESTED** [review]
I've submitted a fews changes to be done, let me know if you want any kind of help 🙂

**Comment by michaelmaillot** [comment]
Hi @nsoeth,

Are you still working on this one? Do you need any kind of help?

**Comment by nsoeth** [comment]
Hi @michaelmaillot 
Sorry, lot of things to do the last few months but I updated the PR according to your review.

**Comment by michaelmaillot** [comment]
Merged manually, thank you!

**Pull request merged**